src/shared/dissect-image.c

```diff
@@ -579,6 +580,28 @@ int dissect_image(
         if (!m)
                 return -ENOMEM;
 
+        r = sd_device_get_sysname(d, &sysname);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to get device sysname: %m");
+        if (startswith(sysname, "loop")) {
+                const char *full_path;
+                _cleanup_free_ char *name_stripped = NULL;
+
+                r = sd_device_get_sysattr_value(d, "loop/backing_file", &full_path);
+                if (r < 0)
+                        log_debug_errno(r, "Failed to lookup image name via loop device backing file sysattr, ignoring: %m");
+
+                r = raw_strip_suffixes(basename(full_path), &name_stripped);
```

Comment:
full_path will be NULL here if the sd_device_get_sysattr_value() call failed → crash

Reply:
oops